document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt

```diff
@@ -43,14 +49,33 @@ import org.springframework.web.reactive.function.client.ExchangeFilterFunctions
 import org.springframework.web.reactive.function.client.ExchangeStrategies
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
 
 class SmartDocumentsClient(
     private var smartDocumentsConnectorProperties: SmartDocumentsConnectorProperties,
     private val smartDocumentsWebClientBuilder: WebClient.Builder,
     private val maxFileSizeMb: Int,
     private val temporaryResourceStorageService: TemporaryResourceStorageService,
+    private val pluginService: PluginService
 ) {
 
+    fun getDocumentStructure(): DocumentsStructure? {
```

Comment:
Its better to use the SmartDocumentsPropertiesDto as an argument in this function so that the PluginService won't need to be injected